hal/src/nRF52840/exflash_hal.cpp

```diff
@@ -0,0 +1,739 @@
+                    cinstr_cfg.length = NRF_QSPI_CINSTR_LEN_1B;
+                } else {
+                    /* Wake up external flash from deep power down mode by sending read id command */
+                    cinstr_cfg.opcode = HAL_QSPI_CMD_STD_READ_ID;
+                    cinstr_cfg.length = NRF_QSPI_CINSTR_LEN_4B;
+                }
+                break;
+            case HAL_EXFLASH_COMMAND_READID:
+                cinstr_cfg.opcode = HAL_QSPI_CMD_STD_READ_ID;
+                cinstr_cfg.length = NRF_QSPI_CINSTR_LEN_4B;
+                break;
+            case HAL_EXFLASH_COMMAND_SUSPEND_PGMERS:
+                /* suspend an ongoing program or erase operation */
+                cinstr_cfg.opcode = flash_params->suspend_opcode;
+                break;
+            case HAL_EXFLASH_COMMAND_RESET:
+                cinstr_cfg.opcode = HAL_QSPI_CMD_STD_RSTEN;
+                break;
+            default:
+                return SYSTEM_ERROR_NOT_SUPPORTED;
+        }
+
+        // Execute the command.
+        CHECK(exflash_qspi_cinstr_xfer(&cinstr_cfg, nullptr, result));
+
+        // For the reset procedure, we need to execute the reset command going forward.
+        if (cmd == HAL_EXFLASH_COMMAND_RESET) {
+            cinstr_cfg.opcode = HAL_QSPI_CMD_STD_RST;
+            CHECK(exflash_qspi_cinstr_xfer(&cinstr_cfg, nullptr, nullptr));
+            switch (flash_type) {
+                case HAL_QSPI_FLASH_TYPE_MX25L3233F: {
+                    CHECK(mx25l32_switch_to_qspi());
+                    break;
+                };
+                case HAL_QSPI_FLASH_TYPE_MX25R6435F: {
+                    CHECK(mx25r64_switch_to_qspi());
+                    break;
+                };
+                case HAL_QSPI_FLASH_TYPE_GD25WQ64E: {
+                    CHECK(gd25_switch_to_qspi());
+                    break;
+                };
+                default:
+                    break;
+            }
+        }
+    } else if (sp == HAL_EXFLASH_SPECIAL_SECTOR_OTP) {
+        // OTP-specific commands
+        switch (cmd) {
+            case HAL_EXFLASH_COMMAND_LOCK_ENTIRE_OTP: {
+                /* Secured OTP Indicator bit (4K-bit Secured OTP)
+                 * 1 = factory lock
+                 * This will block the whole OTP region.
+                 */
+                uint8_t v = 0x01;
+                if (flash_type == HAL_QSPI_FLASH_TYPE_MX25L3233F || flash_type == HAL_QSPI_FLASH_TYPE_MX25R6435F) {
+                    CHECK(exflash_qspi_cinstr_quick_send(HAL_QSPI_CMD_MX25_WRSCUR, get_nrf_cinstr_length(2), &v));
+                } else {
+                    // TODO: Implement for GD25 flash chip
+                    return SYSTEM_ERROR_NOT_SUPPORTED;
+                }
+                break;
+            };
+            case HAL_EXFLASH_COMMAND_GET_OTP_SIZE: {
+                const hal_exflash_params_t* flash_params = hal_exflash_get_params(flash_type);
+                return flash_params->otp_size;
```

Comment:
Is it ok to return the OTP size directly from the function? Or should I set the value to `result`?

Reply:
Probably better to use the result.